consul-core/src/main/java/com/smoketurner/dropwizard/consul/ConsulBundle.java

```diff
@@ -102,9 +106,32 @@ public void initialize(Bootstrap<?> bootstrap) {
         // getConsulAgentHost() and getConsulAgentPort() if Consul is not
         // listening on the default localhost:8500.
         try {
-            final Consul consul = Consul.builder().withHostAndPort(HostAndPort
-                    .fromParts(getConsulAgentHost(), getConsulAgentPort()))
+            final Consul consul;
+
+            if(getConsulAclToken() == null) {
```

Comment:
above this line, why not start creating the `Consul.builder()`, then only if `getConsulAclToken()` isn't an empty `Optional`, you then do the headers stuff below?